src/platform/packages/shared/kbn-unified-tabs/src/components/tab_preview/tab_preview.tsx

```diff
@@ -0,0 +1,181 @@
+  previewQuery: PreviewQuery;
+  stopPreviewOnHover?: boolean;
+  previewDelay?: number;
+}
+
+export const TabPreview: React.FC<TabPreviewProps> = ({
+  children,
+  showPreview,
+  setShowPreview,
+  tabItem,
+  previewQuery,
+  stopPreviewOnHover,
+  previewDelay = 500,
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const [previewTimer, setPreviewTimer] = useState<NodeJS.Timeout | null>(null);
+  const tabRef = useRef<HTMLSpanElement>(null);
+  const [tabPosition, setTabPosition] = useState({ top: 0, left: 0 });
+
+  useEffect(() => {
+    if (showPreview && tabRef.current) {
+      const rect = tabRef.current.getBoundingClientRect();
+      const windowWidth = window.innerWidth;
+
+      // Check if preview would extend beyond right edge
+      const wouldExtendBeyondRight = rect.left + PREVIEW_WIDTH > windowWidth;
+
+      // Calculate left position based on screen edge constraints
+      let leftPosition = rect.left + window.scrollX;
+
+      if (wouldExtendBeyondRight) {
+        // Align right edge of preview with right edge of window
+        leftPosition = windowWidth - PREVIEW_WIDTH + window.scrollX;
+      }
+
+      setTabPosition({
+        top: rect.bottom + window.scrollY,
+        left: leftPosition,
+      });
+    }
+  }, [showPreview]);
+
+  const onKeyDown = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.key === keys.ESCAPE) {
+        setShowPreview(false);
+      }
+    },
+    [setShowPreview]
+  );
+
+  useEvent('keydown', onKeyDown);
+
+  useEffect(() => {
+    return () => {
+      if (previewTimer) {
+        clearTimeout(previewTimer);
+      }
+    };
+  }, [previewTimer]);
+
+  const handleMouseEnter = () => {
+    if (stopPreviewOnHover) return;
+
+    const timer = setTimeout(() => {
+      setShowPreview(true);
+    }, previewDelay);
+
+    setPreviewTimer(timer);
+  };
+
+  const handleMouseLeave = () => {
+    if (previewTimer) {
+      clearTimeout(previewTimer);
+      setPreviewTimer(null);
+    }
+
+    setShowPreview(false);
+  };
+
+  return (
+    <div>
+      <span onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave} ref={tabRef}>
+        {children}
+      </span>
+      <EuiPortal>
+        <EuiSplitPanel.Outer
+          grow
+          css={getPreviewContainerCss(euiTheme, showPreview, tabPosition)}
+          data-test-subj={`unifiedTabs_tabPreview_${tabItem.id}`}
+        >
+          <EuiSplitPanel.Inner paddingSize="none" css={getSplitPanelCss(euiTheme)}>
+            <EuiCodeBlock language={previewQuery.language} transparentBackground paddingSize="none">
+              {previewQuery.query}
+            </EuiCodeBlock>
+          </EuiSplitPanel.Inner>
+          <EuiSplitPanel.Inner
+            grow={false}
+            color="subdued"
+            paddingSize="none"
+            css={getSplitPanelCss(euiTheme)}
+          >
+            {previewQuery.status === TabStatus.RUNNING ? (
+              <EuiFlexGroup alignItems="center" gutterSize="s">
```

Comment:
Small issue when the loading is shown:

<img width="310" alt="Screenshot 2025-03-18 at 17 05 10" src="https://github.com/user-attachments/assets/7f9fa624-b5a4-47c8-9824-501b20724da7" />

Should be solvable with `responsive={false}` on `EuiFlexGroup`.

Reply:
I've added `responsive={false}`, but out of curiosity, do you have a hunch why I can't reproduce that?

<img width="344" alt="Screenshot 2025-03-19 at 12 52 29" src="https://github.com/user-attachments/assets/7079bb60-d919-494e-b4b6-90ac58a8db73" />